repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/versioning/VersioningForIdentityTest.groovy

```diff
@@ -0,0 +1,198 @@
+import groovy.json.JsonSlurper
+import org.rackspace.deproxy.Deproxy
+import spock.lang.Unroll
+
+/**
+ * Created by jennyvo on 5/31/16.
+ *  Versioning for identity
+ */
+class VersioningForIdentityTest extends ReposeValveTest {
+    def static Map acceptXML = [accept: "application/xml"]
+    def static Map acceptJSON = [accept: "application/json"]
+    def static Map acceptV1XML = [accept: 'application/v1+xml']
+    def static Map acceptV2XML = [accept: 'application/v2+xml']
+    def static Map acceptV1JSON = [accept: 'application/v1+json']
+    def static Map acceptV2JSON = [accept: 'application/v2+json']
+    def static Map acceptHtml = [accept: 'text/html']
+    def static Map acceptXHtml = [accept: 'application/xhtml+xml']
+    def static Map acceptXMLWQ = [accept: 'application/xml;q=0.9,*/*;q=0.8']
+    def static Map acceptV2VendorJSON = [accept: 'application/vnd.vendor.service-v2+json']
+    def static Map acceptV2VendorJSON2 = [accept: 'application/vnd.vendor.service+json; version=2']
+    def static Map acceptV2VendorXML = [accept: 'application/vnd.vendor.service+xml; version=2']
+    def static Map acceptV2VendorXML2 = [accept: 'application/vnd.vendor.service-v2+xml']
+    def static Map acceptV1VendorJSON = [accept: 'application/vnd.vendor.service-v1+json']
+    def static Map acceptV1VendorJSON2 = [accept: 'application/vnd.vendor.service+json; version=1']
+    def static Map acceptV1VendorXML = [accept: 'application/vnd.vendor.service+xml; version=1']
+    def static Map acceptV1VendorXML2 = [accept: 'application/vnd.vendor.service-v1+xml']
+
+    def static Map contentXML = ["content-type": "application/xml"]
+    def static Map contentJSON = ["content-type": "application/json"]
+
+    //Start repose once for this particular translation test
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+        deproxy.addEndpoint(properties.targetPort2)
+
+        def params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/versioning/identity", params)
+        repose.start()
+    }
+
+    def cleanupSpec() {
+        deproxy.shutdown()
+        repose.stop()
+    }
+
+    @Unroll
+    def "when retrieving all versions: #reqHeaders"() {
+        when: "User sends requests through repose"
+        def mc = deproxy.makeRequest(url: reposeEndpoint, method: 'GET', headers: reqHeaders)
+
+        then: "Response body should contain"
+        mc.receivedResponse.code == "200"
+
+        for (String st : shouldContain) {
+            mc.receivedResponse.body.contains(st)
+        }
+
+        where:
+        reqHeaders | shouldContain
+        acceptXML  | ['id="/v1"', 'id="/v2"']
+        acceptJSON | ['"id" : "/v1"', '"id" : "/v2"']
+    }
+
+    def "verify the response JSON is formatted for identity when the format is not configured for a versions request"() {
```

Comment:
The method name should be updated to indicate that the format is being configured for the IDENTITY format.